src/litserve/server.py

```diff
@@ -220,30 +223,101 @@ def launch_inference_worker(self, num_uvicorn_servers: int):
-            if len(device) == 1:
-                device = device[0]
-
-            self.workers_setup_status[worker_id] = False
-
-            ctx = mp.get_context("spawn")
-            process = ctx.Process(
-                target=inference_worker,
-                args=(
-                    self.lit_api,
-                    self.lit_spec,
-                    device,
-                    worker_id,
-                    self.request_queue,
-                    self.response_queues,
-                    self.max_batch_size,
-                    self.batch_timeout,
-                    self.stream,
-                    self.workers_setup_status,
-                ),
-            )
-            process.start()
-            process_list.append(process)
+
+        def has_preprocess_method_override(api_instance):
+            # Check if preprocess is implemented by the user
+            if hasattr(api_instance, "preprocess"):
+                return api_instance.preprocess.__code__ is not LitAPI.preprocess.__code__
+            # If preprocess method does not exist in the subclass
+            return False
+
+        enable_process_worker = has_preprocess_method_override(self.lit_api)
+        logger.debug("override lit_api.preprocess method: ", enable_process_worker)
+
+        if enable_process_worker:
+            if self.num_preprocess_workers:
+                pass
+            else:
+                logger.info("Using Workers per device as number of preprocess workers")
+                self.num_preprocess_workers = self.workers_per_device
+
+            self.preprocess_workers = self.devices * self.num_preprocess_workers
+            self.preprocess_queue = manager.Queue()
+
+            for worker_id, device in enumerate(self.preprocess_workers):
+                if len(device) == 1:
+                    device = device[0]
+
+                worker_id = f"preprocess_{worker_id}"
+                ctx = mp.get_context("spawn")
+                process = ctx.Process(
+                    target=preprocess_worker,
+                    args=(
+                        self.lit_api,
+                        self.lit_spec,
+                        device,
+                        worker_id,
+                        self.request_queue,
+                        self.preprocess_queue,
+                        self.response_queues,
+                        self.max_batch_size,
+                        self.batch_timeout,
+                    ),
+                )
+                process.start()
+                process_list.append(process)
+
+            for worker_id, device in enumerate(self.inference_workers):
+                if len(device) == 1:
+                    device = device[0]
+                worker_id = f"inference_{worker_id}"
+                self.workers_setup_status[worker_id] = False
+                ctx = mp.get_context("spawn")
+                process = ctx.Process(
+                    target=inference_worker,
```

Comment:
All good except one concern @lantiga @Usama3059 that if we start a new process with `litapi` then we also have to call `litapi.setup` which will load the model on "non inference worker" process too.

Reply:
@aniketmaurya Based on discussion, we're considering preprocess workers as optional and not requiring any setup for simplicity. The initialized LitAPI will be passed to the preprocess workers, and the setup will only happen on the inference workers, Let me know if I’m missing any information

Reply:
let's consider the following example where we create tokenizer and model in the setup method, now with preprocess enabled it won't be able to access `self.tokenizer` unless we setup the LitAPI in `preprocess` process as well. 

```python
from transformers import AutoTokenizer, AutoModelForSequenceClassification
import torch
from litserve import LitAPI, LitServer

class BERTLitAPI(LitAPI):
    def setup(self, device):
        self.tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
        self.model = AutoModelForSequenceClassification.from_pretrained("bert-base-uncased")
       
    def decode_request(self, request):
        return request["text"]

    def preprocess(self, inputs):
        return self.tokenizer(inputs, return_tensors="pt", padding=True, truncation=True, max_length=512)

    ....

if __name__ == "__main__":
    api = BERTLitAPI()
    server = LitServer(api, accelerator='cuda', devices=1)
    server.run(port=8000)
```

Reply:
@ankitsharma07 Could we provide a warning to the user as shown in the attached image?
![warn](https://github.com/user-attachments/assets/a299f514-cfab-45d7-94fb-34cc59555cd4)

Reply:
@aniketmaurya I guess for simplicity, by 'warning,' I mean it will prevent the `launch_inference_worker` from starting until the code is corrected. We can place the checks after the 'enable_process_worker'.

Reply:
it's a valid concern @aniketmaurya 
we could have an optional `setup_preprocessing` hook as well in the future

Reply:
@aniketmaurya  is there anything else that needs to be worked on for this PR?